File: dcp-solutions/2025-02-05/solution.rb
```ruby
def gray_code(n)
  return [] if n <= 0
  gray = [0]
  (1 << (n - 1)).times do |x|
    gray << x | (1 << (n - 1))
    gray << x ^ (1 << (n - 1))
  end
  gray.sort!
  gray.reverse!
end
```

File: dcp-solutions/2025-02-05/solution.go
```go
package main

import (
	"fmt"
)

func grayCode(n int) []string {
	var grayCode []string
	for i := 0; i < (1 << n); i++ {
		bin := fmt.Sprintf("%b", i)
		gray := ""
		for j := len(bin) - 1; j >= 0; j-- {
			if bin[j] == '0' && gray != "" && gray[gray Len-1] == '1' {
				gray = string(reverse(rune(gray)) + rune(bin[j]))
			} else {
				gray += string(bin[j])
			}
		}
		grayCode = append(grayCode, gray)
	}

	return grayCode
}

func reverse(chars []rune) []rune {
	runes := make([]rune, len(chars))
	for i, r := range chars {
		runes[len(runes)-1-i] = r
	}
	return runes
}

func main() {}
```